vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc

```diff
@@ -41,134 +53,863 @@ See <<PropertyMatchExpression.adoc#top>> for more information.
+
+[example]
+CountMat("G1", "StepCount", "{Unit==true"})
+
+Count the number of pieces in the current Mat/Cargo group and that have a non-blank value for the _StepCount_ property and that have a value of true for the _Unit_ property. +
+
+|===
+
+[#RangeBased]
+==== Range based functions
+Use these functions to *Sum* or *Count* values on pieces that are with a certain range of a source piece, or of a known location.
+
+Each function has two versions. +
+
+The first version assumes the range is in the 'native' format of the *from* location (Grid Units). If a Hex or Square grid exists at that location, then the Range will be assumed to be in appropriate grid cells (hexes or squares). If there is no grid, or an Irregular grid at that location, then the range is assumed to be in pixels. +
 
-For all Math base functions, the asking piece can either be a Mat or a Mat Cargo. If the piece is not a Mat, and/or if it is a Mat Cargo piece not presently on a Mat, then only the piece itself is checked.
+The second *Px* version of the function forces the Range to be checked in Pixels, even if a Hex or Square grid exists at the *From* location.
 
-*SumMat(prop)* returns the total of the specified property among a Mat and its Mat Cargo.
+All Range functions take a minimum and a maximum range as parameters. Pieces will be included in the *Count* or *Sum* if their range from the calling piece is greater than or equals to the minimum range and less than or equal to the maximum range.
 
-*SumMat(prop, expression)* returns the total of the specified property among a Mat and its Mat Cargo. The Mat and each Cargo must also match the supplied Property Expression before being added to the total.
+Error checking is performed and a negative or non-numeric minimum range will be treated as 0. A non-numeric maximum range, or one that is less than the minimum range will be treated as equal to the minimum range.
 
-*CountMat(prop)* returns the number of pieces with a non-blank value for the specified property, among a Mat and its Mat Cargo.
+NOTE: On maps with a <<ZonedGrid.adoc#top,Zoned Grid>>, the Grid at the *From* location is assumed to fill the entire Map for the purposes of the Range calculation.
 
-*CountMat(expression)* returns the number of pieces that match the supplied expression, among a Mat and its Mat Cargo.
+NOTE: If a range-based function is called from within a Piece and a minimum range of 0 is specified, then the calling piece will be included in the *Sum* or *Count* returned.
+
+[#SumRange]
+[width=100%,cols="33%a,67%a"]
+|===
+|*SumRange* |Returns the total of the named property in pieces within range (in Grid Units) of the current piece +
+
+|SumRange(min, max, prop)
+
+#[ Trait Only ]#| Sum a property across all pieces within the specified range in Grid Units +
+
+[example]
+SumRange(0, 0, "Strength") +
+
+Sum the value of the property _Strength_ in all pieces in exactly the same X,Y location as the current piece.
```

Comment:
Should this say "grid location" instead of "X, Y location"?  (If snapping isn't turned on, it's possible for the former to be true without the latter, if I understand correctly.)

Reply:
No, SumRange does exactly what it says. SumRange and CountRange work based on X, Y location, not on GridLocation at all. 0 Range means exactly 0. I've added an additional note to this affect.